src/nlp/Preprocessing.py

```diff
@@ -253,3 +257,75 @@ def correct(self):
       corrected = doc._.outcome_spellCheck
 
     return corrected
+
+  def handleAbbreviations(self, abbrDatabase, type):
+    """
+      Performs automatic correction of abbreviations and returns corrected text
+      This method relies on a database of abbreviations located at:
+      src/nlp/data/abbreviations.xlsx
+      This database contains the most common abbreviations collected from literarture and
+      it provides for each abbreviation its corresponding full word(s); an abbreviation might
+      have multple words associated. In such case the full word that makes more sense given the
+      context is chosen (see findOptimalOption method)
+      @ In, abbrDatabase, pandas dataframe, dataframe containing library of abbreviations
+                                            and their correspoding full expression
+      @ In, type, string, type of abbreviation method ('spellcheck','hard','mixed')
+      @ Out, options, list, list of corrected text options
+    """
+    if type == 'spellcheck':
+      unknowns = self.getMisspelledWords()
+    elif type == 'hard' or type=='mixed':
+      unknowns = []
+      splitSent = self.text.split()
+      for word in splitSent:
+        if word in abbrDatabase['Abbreviation'].values.tolist():
+          unknowns.append(word)
+      if type=='mixed':
+        set1 = set(self.getMisspelledWords())
+        set2 = set(unknowns)
+        unknowns = list(set1.union(set2))
+
+    corrections={}
+    for word in unknowns:
+      if word.lower() in abbrDatabase['Abbreviation'].values:
+        locs = abbrDatabase['Abbreviation'][abbrDatabase['Abbreviation']==word].index.values
+        corrections[word] = abbrDatabase['Full'][locs].values.tolist()
+      else:
+        from difflib import SequenceMatcher
```

Comment:
please add some comments for this comparison

Reply:
comments added